### PR TITLE
add parameter to bubbled revert wrapper

### PR DESCRIPTION
**Pull request by gretzke**
## Related Issue

Which issue does this pull request resolve?

### CODE REVIEW

File: src/libraries/CustomRevert.sol [hunk 1]
```diff
@@ -75,20 +75,21 @@ library CustomRevert {
     }
 
     /// @notice bubble up the revert message returned by a call and revert with the selector provided
-    /// @dev this function should only be used with custom errors of the type `CustomError(bytes revertReason)`
-    function bubbleUpAndRevertWith(bytes4 selector) internal pure {
+    /// @dev this function should only be used with custom errors of the type `CustomError(address contract_, bytes revertReason)`
```
Comment (snreynolds) [inline src/libraries/CustomRevert.sol:78]:
```suggestion
    /// @dev this function should only be used with custom errors of the type `CustomError(address contract, bytes revertReason)`
```

**Review by snreynolds — COMMENTED** [review]
lgtm but I would like @hensha256 to review

File: src/libraries/CustomRevert.sol [hunk 2]
```diff
@@ -75,20 +75,21 @@ library CustomRevert {
...
+    function bubbleUpAndRevertWith(bytes4 selector, address addr) internal pure {
         assembly ("memory-safe") {
             let size := returndatasize()
             let fmp := mload(0x40)
 
             // Encode selector, offset, size, data
```
Comment (hensha256) [inline src/libraries/CustomRevert.sol:84]: this needs updating and maybe a little more detail now
```
// Encode selector, address, data offset, data size, data
```

**Review by hensha256 — COMMENTED** [review]
one nit

File: src/types/Currency.sol [hunk 3]
```diff
@@ -31,12 +31,12 @@ library CurrencyLibrary {
...
-    /// @param revertReason bubbled up revert reason
-    error NativeTransferFailed(bytes revertReason);
+    /// @param reason bubbled up revert reason
+    error Wrap__NativeTransferFailed(address recipient, bytes reason);
 
     /// @notice Thrown when an ERC20 transfer fails
-    /// @param revertReason bubbled up revert reason
-    error ERC20TransferFailed(bytes revertReason);
+    /// @param reason bubbled up revert reason
+    error Wrap__ERC20TransferFailed(address token, address recipient, bytes reason);
```
Comment (hensha256) [inline src/types/Currency.sol:39]: is this right? looks like its only passing in 1 address not two

File: src/libraries/CustomRevert.sol (hunk 1 above)
Comment (gretzke) [reply src/libraries/CustomRevert.sol:78]: but `contract` is a reserved keyword 😄

File: src/types/Currency.sol (hunk 3 above)
Comment (gretzke) [reply src/types/Currency.sol:39]: ugh, thanks for catching it!

File: src/libraries/CustomRevert.sol [hunk 4]
```diff
@@ -75,13 +75,13 @@ library CustomRevert {
     }
 
     /// @notice bubble up the revert message returned by a call and revert with the selector provided
-    /// @dev this function should only be used with custom errors of the type `CustomError(address contract_, bytes revertReason)`
+    /// @dev this function should only be used with custom errors of the type `CustomError(address contract, bytes revertReason)`
```
Comment (hensha256) [inline src/libraries/CustomRevert.sol:78]: i actually feel like `contract` is wrong? for our `Wrap__NativeTransferFailed` the `recipient` thats emitted might not be a contract. Maybe `target` is better? And that avoids the contract keyword too

Comment (gretzke) [reply src/libraries/CustomRevert.sol:78]: target is good, I like that

Comment (snreynolds) [reply src/libraries/CustomRevert.sol:78]: i like target!

**Pull request merged**